src/adapter/threads.ts

```diff
@@ -572,18 +572,41 @@ export class Thread implements VariableStoreDelegate {
     };
   }
 
-  rawLocationToUiLocation(rawLocation: RawLocation): Promise<PreferredUILocation | undefined> {
-    const script = rawLocation.scriptId ? this._scripts.get(rawLocation.scriptId) : undefined;
-    if (!script)
-      return Promise.resolve(undefined);
-    let {lineNumber, columnNumber} = rawToUiOffset(rawLocation, this.defaultScriptOffset());
+  public async rawLocationToUiLocation(rawLocation: RawLocation): Promise<PreferredUILocation | undefined> {
+    const script = rawLocation.scriptId ? await this.getScriptByIdOrWait(rawLocation.scriptId) : undefined;
+    if (!script) {
+      return;
+    }
+
+    const { lineNumber, columnNumber } = rawToUiOffset(rawLocation, this.defaultScriptOffset());
     return this._sourceContainer.preferredUiLocation({
       lineNumber,
       columnNumber,
       source: script.source
     });
   }
 
+  /**
+   * Gets a script ID if it exists, or waits to up maxTime. In rare cases we
+   * can get a request (like a stacktrace request) from DAP before Chrome
+   * finishes passing its sources over. We *should* normally know about all
+   * possible script IDs; this waits if we see one that we don't.
+   */
+  private async getScriptByIdOrWait(scriptId: string, maxTime: number = 500) {
```

Comment:
We were getting this sometimes in the stacktrace-related tests. I think this is a simple and safe solution.